### PR TITLE
Distingiush between signature algorithm with and without parameter.

### PR DESCRIPTION
**Pull request by partim**
This PR adjusts certificate and signed object validation to follow OpenSSL’s practice of refusing certificates with mismatching encoding of the signature algorithm inside and outside the signed portion.

### CODE REVIEW

File: src/repository/cert.rs [hunk 1]
```diff
@@ -2646,6 +2646,33 @@ mod test {
...
+            include_bytes!(
+                "../../test-data/signature-alg-mismatch.mft"
+            ).as_ref(),
+            false
+        ).unwrap();
+        assert!(
+            mft.cert().inspect_ee_at(
+                true, Time::utc(2020, 05, 01, 0, 0, 0)
+            ).is_err()
+        );
```
Comment (DRiKE) [inline src/repository/cert.rs:2673]: Is this testing the new with/without parameter support added in this PR, or is this testing for a possible malformed error (as returned on line 123)? If possible/trivial, being more specific than just `.is_err()` could sort of self-document this test a bit better, and make it more robust as well. (Even though the mft filename hints at what is going on here)

**Review by DRiKE — APPROVED** [review]
Small clarification on the test would help, but that's not a blocker IMO.

File: src/repository/cert.rs (hunk 1 above)
Comment (partim) [reply src/repository/cert.rs:2673]: Given that the error is really a unit struct, `is_err` should be sufficient here. I will add a comment, though.

Comment (DRiKE) [reply src/repository/cert.rs:2679]: That helps a lot, thanks!

**Pull request merged**